cmd/chaininfo/chain_info.go

```diff
@@ -18,64 +21,85 @@ import (
 	ChainParameters *json.RawMessage    `json:"chain-parameters"`
 	ChainConfig     *params.ChainConfig `json:"chain-config"`
 	RollupAddresses *RollupAddresses    `json:"rollup"`
 }
 
-func GetChainConfig(chainId *big.Int, genesisBlockNum uint64, l2ChainInfoFiles []string) (*params.ChainConfig, error) {
-	chainInfo, err := ProcessChainInfo(chainId.Uint64(), l2ChainInfoFiles)
+func GetChainConfig(chainId *big.Int, chainName string, genesisBlockNum uint64, l2ChainInfoFiles []string) (*params.ChainConfig, error) {
+	chainInfo, err := ProcessChainInfo(chainId.Uint64(), chainName, l2ChainInfoFiles)
 	if err != nil {
 		return nil, err
 	}
 	if chainInfo.ChainConfig != nil {
 		chainInfo.ChainConfig.ArbitrumChainParams.GenesisBlockNum = genesisBlockNum
 		return chainInfo.ChainConfig, nil
 	}
-	return nil, fmt.Errorf("missing chain config for L2 chain ID %v", chainId)
+	if chainId.Uint64() != 0 {
+		return nil, fmt.Errorf("missing chain config for L2 chain ID %v", chainId)
+	} else {
+		return nil, fmt.Errorf("missing chain config for L2 chain name %v", chainName)
+	}
 }
 
-func GetRollupAddressesConfig(chainId *big.Int, l2ChainInfoFiles []string) (RollupAddresses, error) {
-	chainInfo, err := ProcessChainInfo(chainId.Uint64(), l2ChainInfoFiles)
+func GetRollupAddressesConfig(chainId uint64, chainName string, l2ChainInfoFiles []string) (RollupAddresses, error) {
+	chainInfo, err := ProcessChainInfo(chainId, chainName, l2ChainInfoFiles)
 	if err != nil {
 		return RollupAddresses{}, err
 	}
 	if chainInfo.RollupAddresses != nil {
 		return *chainInfo.RollupAddresses, nil
 	}
-	return RollupAddresses{}, fmt.Errorf("missing rollup addresses for L2 chain ID %v", chainId)
+	if chainId != 0 {
+		return RollupAddresses{}, fmt.Errorf("missing rollup addresses for L2 chain ID %v", chainId)
+	} else {
+		return RollupAddresses{}, fmt.Errorf("missing rollup addresses for L2 chain name %v", chainName)
+	}
 }
 
-func ProcessChainInfo(chainId uint64, l2ChainInfoFiles []string) (*ChainInfo, error) {
+func ProcessChainInfo(chainId uint64, chainName string, l2ChainInfoFiles []string) (*ChainInfo, error) {
 	for _, l2ChainInfoFile := range l2ChainInfoFiles {
 		chainsInfoBytes, err := os.ReadFile(l2ChainInfoFile)
 		if err != nil {
 			return nil, fmt.Errorf("failed to read file %s err %w", l2ChainInfoFile, err)
 		}
-		var chainsInfo map[uint64]ChainInfo
+		var chainsInfo []ChainInfo
 		err = json.Unmarshal(chainsInfoBytes, &chainsInfo)
 		if err != nil {
-			return nil, err
+			decodedChainsInfoBytes, err := io.ReadAll(base64.NewDecoder(base64.StdEncoding, bytes.NewReader(chainsInfoBytes)))
+			if err != nil {
+				return nil, err
+			}
+			err = json.Unmarshal(decodedChainsInfoBytes, &chainsInfo)
+			if err != nil {
+				return nil, err
+			}
 		}
-		if _, ok := chainsInfo[chainId]; !ok {
-			continue
+		for _, chainInfo := range chainsInfo {
+			if chainInfo.ChainId == chainId || chainInfo.ChainName == chainName {
```

Comment:
If both chainId and chainName are specified (meaning they're nonzero and nonempty respectively) we should probably require that they both match

Reply:
done.